File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.js
function solution(r1, r2) {
    let answer = 0;

    for (let i = 1; i < r2; i++) {
        let num2 = Math.floor(Math.sqrt(r2 * r2 - i * i));
        if (i >= r1) answer += 4 * (num2 + 1);
        else {
            let num1 = Math.floor(Math.sqrt(r1 * r1 - i * i));
            if (num1 * num1 + i * i === r1 * r1) answer += 4 * (num2 - num1 + 1);
            else answer += 4 * (num2 - num1);
        }
    }

    return answer + 4;
}

File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.py
def solution(r1, r2):
    answer = 0
    for i in range(0, r1):
        answer += int((r2**2 - i**2) ** 0.5) - int((r1**2 - i**2 - 1) ** 0.5)
    for i in range(r1, r2):
        answer += int((r2**2 - i**2) ** 0.5)
    return answer * 4